### PR TITLE
making load_hosp_data more general, to do flu or covid.

### PR DESCRIPTION
**Pull request by nickreich**
Note that I created load_hosp_data() as the workhorse, but also left a wrapper for load_flu_hosp_data() that should work for backwards compatibility.

### CODE REVIEW

**Comment by nickreich** [comment]
```
tmp_flu <- load_hosp_data(pathogen="flu")
tmp_covid <- load_hosp_data(pathogen="covid")
tmp_flu_covidcast <- load_hosp_data(pathogen="flu", source="covidcast")
tmp_covid_covidcast <- load_hosp_data(pathogen="covid", source="covidcast")

flu_hhs <- ggplot(filter(tmp_flu, location_name == "Massachusetts"), aes(x=date, y=value)) +
  geom_point() + geom_line() + coord_cartesian(xlim=c(as.Date("2021-12-15"), as.Date("2022-01-15"))) +
  ggtitle("flu hhs")

covid_hhs <- ggplot(filter(tmp_covid, location_name == "Massachusetts"), aes(x=date, y=value)) +
  geom_point() + geom_line() + coord_cartesian(xlim=c(as.Date("2021-12-15"), as.Date("2022-01-15")))+
  ggtitle("covid hhs")

flu_cc <- ggplot(filter(tmp_flu_covidcast, location_name == "Massachusetts"), aes(x=date, y=value)) +
  geom_point() + geom_line() + coord_cartesian(xlim=c(as.Date("2021-12-15"), as.Date("2022-01-15")))+
  ggtitle("flu covidcast")

covid_cc <- ggplot(filter(tmp_covid_covidcast, location_name == "Massachusetts"), aes(x=date, y=value)) +
  geom_point() + geom_line() + coord_cartesian(xlim=c(as.Date("2021-12-15"), as.Date("2022-01-15")))+
  ggtitle("covid covidcast")

gridExtra::grid.arrange(flu_hhs, covid_hhs, flu_cc, covid_cc, nrow=2)
```
Yields this:
image

File: R/load_flu_hosp_data.R [hunk 1]
```diff
@@ -91,7 +98,9 @@ load_flu_hosp_data <- function(as_of = NULL,
       jsonlite::fromJSON()
     csv_path <- tail(temp$archive_link$url, 1)
     data <- readr::read_csv(csv_path)
-    state_dat <- data %>%
+    ## value depends on which pathogen
```
Comment (elray1) [inline R/load_flu_hosp_data.R]: I am wondering if we could reduce some code duplication by conditioning inside of the transmute? Am I missing something here?

```suggestion
    ## value depends on which pathogen
    state_dat <- data %>%
        dplyr::transmute(
          geo_value = tolower(state),
          date = date - 1,
          epiyear = lubridate::epiyear(date),
          epiweek = lubridate::epiweek(date),
          value = ifelse(
            pathogen == "flu",
            previous_day_admission_influenza_confirmed,
            previous_day_admission_adult_covid_confirmed + previous_day_admission_pediatric_covid_confirmed)
        ) %>%
        dplyr::arrange(geo_value, date)
```

File: R/load_flu_hosp_data.R [hunk 2]
```diff
@@ -100,9 +109,21 @@ load_flu_hosp_data <- function(as_of = NULL,
...
       ) %>%
       dplyr::arrange(geo_value, date)
+    } else {
+      state_dat <- data %>%
+        dplyr::transmute(
+          geo_value = tolower(state),
+          date = date - 1,
+          epiyear = lubridate::epiyear(date),
+          epiweek = lubridate::epiweek(date),
+          value = previous_day_admission_influenza_confirmed,
```
Comment (elray1) [inline R/load_flu_hosp_data.R]: at a minimum, we could delete this right?

File: R/load_flu_hosp_data.R [hunk 3]
```diff
@@ -142,6 +163,37 @@ load_flu_hosp_data <- function(as_of = NULL,
...
+#' @return data frame of flu incidence with columns date, location,
+#'   location_name, value
+load_flu_hosp_data <- function(as_of = NULL,
+                               locations = "*",
+                               temporal_resolution = "daily",
+                               source = "HealthData",
+                               na.rm = FALSE) {
+  load_hosp_data(pathogen = "flu",
+                 as_of = as_of,
+                 temporal_resolution = temporal_resolution,
```
Comment (elray1) [inline R/load_flu_hosp_data.R:203]: Think we need to add locations parameter in here
```suggestion
                 locations = locations,
                 temporal_resolution = temporal_resolution,
```

File: R/load_flu_hosp_data.R [hunk 4]
```diff
@@ -66,14 +71,16 @@ load_flu_hosp_data <- function(as_of = NULL,
   } else {
     locations_to_fetch <- locations
   }
-  
+
+  signal <- ifelse(pathogen=="flu", "confirmed_admissions_influenza_1d", "confirmed_admissions_covid_1d")
```
Comment (elray1) [inline R/load_flu_hosp_data.R]: maybe move this inside of the `if (source == "covidcast")` block since I think it's only used in there?

**Review by elray1 — CHANGES_REQUESTED** [review]
a few minor comments

File: R/load_flu_hosp_data.R (hunk 1 above)
Comment (nickreich) [reply R/load_flu_hosp_data.R]: Agree, but I tried to do this for over 30 minutes when originally writing it and didn't get it to work, so I gave up.

File: R/load_flu_hosp_data.R (hunk 4 above)
Comment (nickreich) [reply R/load_flu_hosp_data.R]: yup. updating this.

File: R/load_flu_hosp_data.R (hunk 2 above)
Comment (nickreich) [reply R/load_flu_hosp_data.R]: yeah, I'm fixing that.

File: R/load_flu_hosp_data.R (hunk 3 above)
Comment (nickreich) [reply R/load_flu_hosp_data.R:203]: I'll add this too.

**Pull request merged**